client/dashboard/app/queries/geolocation.ts

```diff
@@ -0,0 +1,20 @@
+import { useQuery } from '@tanstack/react-query';
+
+export interface GeoLocationData {
+	city: string;
+	country_long: string;
+	country_short: string;
+	latitude: string;
+	longitude: string;
+	region: string;
+}
+
+export const useGeoLocationQuery = ( options = {} ) =>
+	useQuery< GeoLocationData >( {
```

Comment:
We should only define `queryOptions()` here and not actually calling `useQuery()` in this query layer.